Mochi/README.md

```diff
@@ -45,94 +45,77 @@ On the spacecraft (transfer the binary there)
 - exit: stop the server and exit the program (does not save the trained models).
 
-### Training
-The given training data is specific to the OrbitAI experiment. A generic solution is possible but not in the scope of this experiment for the sake of rapid prototyping.
-- The `train` command takes 8 parameters: the expected label, the photodiode elevation angle values, the photodiode data acquisition timestamp (milliseconds) e.g.: 
+#### Training, Continue Training, and Inferring
+An ML Server implemented by the OrbitAI serves as a generic entry point that accepts training or inference inputs of any size if the following structure:
+- The first four characters is an integer value representing the total length of the message sent to the ML Server.
+- The remaining characters compose the input string sent to the update (train) or predict (infer) functions implemented in the [MochiMochi online ML algorithms](https://github.com/georgeslabreche/MochiMochi).
+
+E.g. training or inference strings expected by the ML Server:
 ```
-train 1 0.11 0.22 0.33 0.44 0.55 1.43 1615255005626
-train 0 0.11 0.22 0.33 0.44 0.55 0.43 1615255010842
+0041 +1 1:1.232 2:2.412 3:2.123 4:5.23223
+0031 -1 1:1.232 2:2.412 3:2.123
 ```
-- Label values can either be 1 or 0 for binary classification. Label values of -1 are interpreted as 0.
-- Expected photodiode elevation angles values are PD6 for Camera and Optical RX FDIR model training and PD3 for Star Tracker model training.
-= The photodiode acquisition timestamp is only passed for logging purposes, it is not used for training.
 
+Note:
+- The label values can either be +1 or -1 for binary classification. These label values represent the expected labels (supervised learning).
+- The ML Server determines which mode its on from the reading the properties file (Mode 0: New Training, Mode 1: Continue Training, Mode 2: Inference).
 ### Test the ML Server
-
 #### Training
 ##### Single Sample
-1. Start the Online ML server: `./OrbitAI_Mochi`
-2. Connect to the ML server: `telnet localhost 9999`
-3. Train the models: `train 1 0.11 0.22 0.33 0.44 0.55 1.43 1615255005626`
-4. Save the models: `save`
+1. Update the properties file to training mode (mode 0).
+2. Start the Online ML server: `./OrbitAI_Mochi <path_to_properties_file>`
+3. Connect to the ML server: `telnet localhost 9999`
+4. Train the models: `0041 +1 1:1.232 2:2.412 3:2.123 4:5.23223`
+5. The models are serialized and saved in the `models` directory.
 
 ##### Batch Samples
-1. Start the Online ML server: `./OrbitAI_Mochi`
-2. Send telnet commands to the server:
+1. Update the properties file to training mode (mode 0).
+2. Start the Online ML server: `./OrbitAI_Mochi <path_to_properties_file>`
+3. Send telnet commands to the server:
 ```
-eval 'input="test_data/camera_tiny.txt"; echo "reset"; while IFS= read -r line; do sleep 0.02; echo "train ${line/+/''} $(date +%s000)"; done < "$input";' | telnet localhost 9999
+TBD
```

Comment:
Because the new ML Server doesn't expect "train" and "infer" prefixes in the command strings. These are determined from the config file. Also we now have to start the command string with the message length. I have to updated the test data and bash scripts to account for these changes. I made it a separate issue: https://github.com/georgeslabreche/opssat-orbitai/issues/47